code/Reports/BrokenFilesReport.php

```diff
@@ -12,9 +12,11 @@
 use SilverStripe\ORM\DB;
 use SilverStripe\Versioned\Versioned;
 
+// phpcs:disable PSR1.Files.SideEffects
```

Comment:
I think it'd be better to put it into phpcs.xml.dist as an exclusion

Reply:
Just had a look at the [PSR1 standard](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-1-basic-coding-standard.md#23-side-effects).

Looks like this is the preferred way of achieving what we are aiming for.

```php
<?php
if (class_exists(Report::class)) {
    class RecentlyEditedReport extends Report
    {
        // ...
    }
}
```

Reply:
Interesting. Kind of goes against the "early return" stuff. But this stuff is all so magical anyway (class definitions are parsed before code is evaluated) - but PHP handles this somehow.

Reply:
I totally disagree with wrapping the entire definition in an if statement, hopefully that's not just me =)

Reply:
It does look inelegant to my eye. However, I would prefer this to disabling `PSR1.Files.SideEffects` across the board.

Reply:
I guess if it's already global in framework - keep it consistent firstly.

However, I'm keen for this change to be localised and "early return" practice :)

Reply:
I'd still really prefer to see these inline rules removed

Reply:
Could we not have some way of only loading a class using the module api otherwise the manifest ignores it.

cms.yml
```
--
Name: cms-reports
Except:
  moduleexists: 'silverstripe/reports'
--
SilverStripe\Core\Manifest\ManifestFileFinder:
  dontAcceptDirs: 
     - 'src/Reports'
```

Reply:
Option 3 because having `if (class_exists()) return` is just so hacky in PHP. Function definitions prefer wrapping anyway... Having something in config to exclude directories would be a blocker for moving to PSR-4 autoloading.

Reply:
Option 1 is the current standard across all core modules